File: 0345-reverse-vowels-of-a-string/0345-reverse-vowels-of-a-string.go
import (
    "strings"
)

func reverseVowels(s string) string {
    const vowels = "aeiouAEIOU"
    sSlice := strings.Split(s, "")
    
    left, right := 0, len(sSlice)-1
    for left < right {
        for left < right && !strings.Contains(vowels, sSlice[left]) {
            left++
        }
        for right > left && !strings.Contains(vowels, sSlice[right]) {
            right--
        }
        sSlice[left], sSlice[right] = sSlice[right], sSlice[left]
        left++
        right--
    } 
    return strings.Join(sSlice, "")
}

File: 0345-reverse-vowels-of-a-string/0345-reverse-vowels-of-a-string.c
char * reverseVowels(char * s){
    int left, right;
    char tmp;
    const char *vowels = "aeiouAEIOU";
    
    left = 0;
    right = strlen(s)-1;
    
    while (left < right) {
        while (left < right && !strchr(vowels, s[left]))
            left++;
        while (right > left && !strchr(vowels, s[right]))
            right--;
        
        tmp = s[left];
        s[left++] = s[right];
        s[right--] = tmp;
    }
    return s;
}